File: python/leetcode/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
	def removeElements(self, head, val):
		root_node = ListNode(0)
		root_node.next = head
		prev_head = root_node
		while prev_head.next is not None:
			if prev_head.next.val == val:
				prev_head.next = prev_head.next.next
			else:
				prev_head = prev_head.next
		return root_node.next

File: java/leetcode/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
    	ListNode rootNode = new ListNode(0);
    	rootNode.next = head;
    	ListNode prevHead = rootNode;
    	while (prevHead.next != null) {
    		if (prevHead.next.val == val) {
    			prevHead.next = prevHead.next.next;
    		} else {
    			prevHead = prevHead.next;
    		}
    	}
    	return rootNode.next;
    }
}
